platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/GradleEnterpriseAutoAppliedPluginRegistry.java

```diff
@@ -50,14 +50,19 @@ public PluginRequests getAutoAppliedPlugins(Settings target) {
         if (((StartParameterInternal) target.getStartParameter()).isUseEmptySettings() || !shouldApplyDevelocityPlugin(target)) {
             return PluginRequests.EMPTY;
         } else {
+            // We are going with an auto-application request, let's configure the URL
+            // TODO Remove this once the default applied version supports DefaultGradleEnterprisePluginConfig.getDevelocityUrl()
+            target.getPluginManager().withPlugin(AutoAppliedDevelocityPlugin.ID.getId(), new DevelocityAutoAppliedPluginConfigurationAction(target));
             return PluginRequests.of(createDevelocityPluginRequest());
         }
     }
 
     private static boolean shouldApplyDevelocityPlugin(Settings settings) {
         Gradle gradle = settings.getGradle();
-        StartParameter startParameter = gradle.getStartParameter();
-        return startParameter.isBuildScan() && gradle.getParent() == null;
+        StartParameterInternal startParameter = (StartParameterInternal) gradle.getStartParameter();
+        return (startParameter.isBuildScan()
+                || (!Strings.isNullOrEmpty(startParameter.getDevelocityUrl()) && !startParameter.isNoBuildScan()))
+            && gradle.getParent() == null;
```

Comment:
I think we should still apply the Develocity plugin even if you pass --no-scan. So we keep the classpath the same and don't have to re-configure everything just because we pass --no-scan.

Reply:
AFAICT both options have the same user facing behavior.
In both cases, you get a CC miss when the only different in an invocation is `--no-scan` being present or not.

But I understand the point about applying the plugin since you are effectively requesting it when passing the Develocity URL option.

Will update the PR.